embedder.py

```diff
@@ -0,0 +1,66 @@
+"""
+Created on 2022 Jan 28 14:09 
+
+@author: keller
+"""
+import abc
+
+import torch
+import ruclip
+import numpy as np
+
+from PIL import Image
+
+from numbers import Number
+from typing import List
+
+class Embedder(abc.ABC):
+    @abc.abstractmethod
+    def encode_text(self, text):
+        pass
+
+    @abc.abstractmethod
+    def encode_imgs(self, imgs):
+        pass
+
+    def cos(self, text_emb: np.ndarray, img_emb: np.ndarray) -> Number:
+        """
+        Returns cos similarity between text_emb and img_emb
+        :param text_emb: 1D tensor
+        :param img_emb: 1D tensor
+        :return: cos similarity (Number)
+        """
+        return np.dot(text_emb, img_emb)
```

Comment:
It's like a dot product, not cos distance

Reply:
That's right. RuCLIP outputs embeddings with norm 1, so I was thinking to simplify the code. But I will add division by norms product, it costs nothing.